runtime/finalise.c

```diff
@@ -202,12 +212,16 @@ void caml_final_do_calls (void)
 #if defined(NATIVE_CODE) && defined(WITH_SPACETIME)
       caml_spacetime_trie_node_ptr = saved_spacetime_trie_node_ptr;
 #endif
-      running_finalisation_function = 0;
+      caml_final_enabled = 1;
       if (Is_exception_result (res)) caml_raise (Extract_exception (res));
     }
     caml_gc_message (0x80, "Done calling finalisation functions.\n");
     if (caml_finalise_end_hook != NULL) (*caml_finalise_end_hook) ();
   }
+  caml_final_to_do = 0;
+#ifdef NATIVE_CODE
+  caml_update_young_limit();
```

Comment:
My first guess was be that you call `update_young_limit` because `caml_final_to_do` has changed, but if I understand correctly, a change from 1 to 0 means that `caml_update_young_limit` will not do anything. So I would guess that this call here is not due to the change in this PR, but you think that it should have been there from the start (due to other conditions that could be triggered by the finalizers run by this function).

Reply:
I have just created an enum for factorizing both flags and have clearer names.